### PR TITLE
[User management] Some cleanup to ongoing-stream-authorization [DPP-830]

### PR DESCRIPTION
**Pull request by nmarton-da**
* Split out UserRightsChecker from OngoingAuthorizationObserver
* Small refactorings

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/OngoingAuthorizationObserver.scala [hunk 1]
```diff
@@ -44,133 +38,122 @@ private[auth] final class OngoingAuthorizationObserver[A](
...
-          onError(staleStreamAuthError)
-        case Success(Right(userRights)) =>
-          val updatedClaims = AuthorizationInterceptor.convertUserRightsToClaims(userRights)
-          if (updatedClaims.toSet != originalClaims.claims.toSet) {
-            onError(staleStreamAuthError)
-          }
-          lastUserRightsCheckTime = nowF()
-      }
-  }
+  override def isCancelled: Boolean = synchronized(observer.isCancelled)
```
Comment (nmarton-da) [inline ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/OngoingAuthorizationObserver.scala:44]: switching to synchronized all over the place (to not have assumptions which part need and not need synchronization)

File: ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/UserRightsChecker.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+)(implicit ec: ExecutionContext)
+    extends UserRightsChecker {
+
+  def schedule(userClaimsMismatchCallback: () => Unit): () => Unit = {
+    if (originalClaims.resolvedFromUser) {
+      // Scheduling a task that periodically checks
+      // whether user rights state has changed.
+      // If user rights state has changed it aborts the stream by calling [[userClaimsMismatchCallback]]
+      val delay = userRightsCheckIntervalInSeconds.seconds
+      val userId = originalClaims.applicationId.fold[Ref.UserId](
```
Comment (nmarton-da) [inline ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/UserRightsChecker.scala:38]: this is no fails early, not sure if that is entirely right

File: ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/OngoingAuthorizationObserver.scala (hunk 1 above)
Comment (pbatko-da) [reply ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/OngoingAuthorizationObserver.scala:44]: These originally unsychronized members are, afaict from the doc strings, either safe to call from mutliple threads or they can be called only early in the lifecycle (i.e. ~ before the stream actually starts)

**Review by pbatko-da — APPROVED** [review]

**Review by mziolekda — APPROVED** [review]
Nice rafctoring

**Pull request merged**